lib/spack/spack/installer.py

```diff
@@ -1634,9 +1639,15 @@ def install(self):
                 self._update_failed(task, True, exc)
                 spack.hooks.on_install_failure(task.request.pkg.spec)
 
+                # Collect short versions of exceptions
+                if isinstance(exc, SpackError):
+                    install_errors.append((pkg.name, SpackError(exc.message)))
+                else:
+                    install_errors.append((pkg.name, exc))
```

Comment:
Possible simplification:

```suggestion
                # Collect short versions of Spack exceptions
                err = SpackError(exc.message) if isinstance(exc, SpackError) else exc
                install_errors.append((pkg.name, err))
```